### ISSUE TITLE
Exposure iop with masks export different results CPU/GPU

### ISSUE THREAD

**Issue by groutr** [darktable-org/darktable#3629]
**Describe the bug**
Applying the xmp file below to any raw file will give different results if opencl is enabled/disabled.

**To Reproduce**
1. Save xmp below to a file
2. Execute `darktable-cli <input_img> <xmpfile> <output_img> --core --conf opencl=true`
and `darktable-cli <input_img> <xmpfile> <output_img> --core --conf opencl=false`
3. Observe that the exposures of the outputs differ noticeably.

**Expected behavior**
I expect both outputs to be the same.

**Screenshots**
With CPU:
[![TEST-CPU-cpu.jpg](https://i.postimg.cc/V6s7cPQw/TEST-CPU-cpu.jpg)](https://postimg.cc/v1j7WjZ2)

With GPU:
[![TEST-GPU.jpg](https://i.postimg.cc/SQJBXC3h/TEST-GPU.jpg)](https://postimg.cc/YjBnZGtn)

**Platform (please complete the following information):**
 - OS: archlinux
 - Version 3.0.0rc1+102~ge9963abec
 - intel 520 w/ NEO opencl libraries

**Additional context**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<x:xmpmeta xmlns:x="adobe:ns:meta/" x:xmptk="XMP Core 4.4.0-Exiv2">
 <rdf:RDF xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#">
  <rdf:Description rdf:about=""
    xmlns:exif="http://ns.adobe.com/exif/1.0/"
    xmlns:xmp="http://ns.adobe.com/xap/1.0/"
    xmlns:xmpMM="http://ns.adobe.com/xap/1.0/mm/"
    xmlns:dc="http://purl.org/dc/elements/1.1/"
    xmlns:darktable="http://darktable.sf.net/"
   xmp:Rating="4"
   darktable:xmp_version="3"
   darktable:raw_params="0"
   darktable:auto_presets_applied="1"
   darktable:history_end="3"
   darktable:iop_order_version="5">
   <darktable:masks_history>
    <rdf:Seq>
     <rdf:li
      darktable:mask_num="0"
      darktable:mask_id="1572019312"
      darktable:mask_type="16"
      darktable:mask_name="gradient #3"
      darktable:mask_version="4"
      darktable:mask_points="1779f63ea7dcfd3e02e486430000003f00000000"
      darktable:mask_nb="1"
      darktable:mask_src="0000000000000000"/>
     <rdf:li
      darktable:mask_num="0"
      darktable:mask_id="1572019318"
      darktable:mask_type="4"
      darktable:mask_name="grp local contrast"
      darktable:mask_version="4"
      darktable:mask_points="701cb35d761cb35d030000000000803f"
      darktable:mask_nb="1"
      darktable:mask_src="0000000000000000"/>
     <rdf:li
      darktable:mask_num="0"
      darktable:mask_id="1572020477"
      darktable:mask_type="16"
      darktable:mask_name="gradient #2"
      darktable:mask_version="4"
      darktable:mask_points="501d933e3f03fe3e2dcf86c30000003f00000000"
      darktable:mask_nb="1"
      darktable:mask_src="0000000000000000"/>
     <rdf:li
      darktable:mask_num="0"
      darktable:mask_id="1572020481"
      darktable:mask_type="4"
      darktable:mask_name="grp exposure 1"
      darktable:mask_version="4"
      darktable:mask_points="fd20b35d0121b35d030000000000803f"
      darktable:mask_nb="1"
      darktable:mask_src="0000000000000000"/>
    </rdf:Seq>
   </darktable:masks_history>
   <darktable:history>
    <rdf:Seq>
     <rdf:li
      darktable:num="0"
      darktable:operation="mask_manager"
      darktable:enabled="0"
      darktable:modversion="2"
      darktable:params="00000000"
      darktable:multi_name=""
      darktable:multi_priority="0"
      darktable:iop_order="22.0000000000000"
      darktable:blendop_version="9"
      darktable:blendop_params="gz11eJxjYGBgkGAAgRNODGiAEV0AJ2iwh+CRyscOAAdeGQQ="/>
     <rdf:li
      darktable:num="1"
      darktable:operation="rawprepare"
      darktable:enabled="1"
      darktable:modversion="1"
      darktable:params="000000000000000000000000000000000000000000000000143e0000"
      darktable:multi_name=""
      darktable:multi_priority="0"
      darktable:iop_order="1.0000000000000"
      darktable:blendop_version="9"
      darktable:blendop_params="gz11eJxjYGBgkGAAgRNODGiAEV0AJ2iwh+CRyscOAAdeGQQ="/>
     <rdf:li
      darktable:num="2"
      darktable:operation="exposure"
      darktable:enabled="1"
      darktable:modversion="5"
      darktable:params="0000000020a089bbb11c5a3f00004842000080c0"
      darktable:multi_name="1"
      darktable:multi_priority="0"
      darktable:iop_order="21.5000000000000"
      darktable:blendop_version="9"
      darktable:blendop_params="gz06eJxjZ2BgYGMAgQwnEMmouDlWAMyf4sgIJM+eyXE48nWF3YUzPnYMOEGDPQSTx1+5YoXtzh0WduTqH3g+dgAA4qUkWQ=="/>
    </rdf:Seq>
   </darktable:history>
  </rdf:Description>
 </rdf:RDF>
</x:xmpmeta>
```

**Comment by upegelow** [comment]
Confirmed. CPU and GPU have different results if mask opacity and/or mask contrast in the refinement feature are non-zero.

![Screenshot_20191130_092331](https://user-images.githubusercontent.com/1794020/69897966-2f992380-1353-11ea-99b3-6e36d610cfd5.png)

**Comment by upegelow** [comment]
should be fixed by PR #3635

**Comment by groutr** [comment]
fix verified.  Thanks @upegelow!
